sdk/core/src/CommonNetworkMember/NetworkMemberWithCognito.ts

```diff
@@ -47,84 +46,56 @@ export abstract class NetworkMemberWithCognito extends BaseNetworkMember {
   constructor(
     password: string,
     encryptedSeed: string,
-    platformEncryptionTools: IPlatformEncryptionTools,
-    affinidiCommon: AffinidiCommonConstructor | null,
+    platformCryptographyTools: IPlatformCryptographyTools,
     options: ParsedOptions,
     component: EventComponent,
     cognitoUserTokens: CognitoUserTokens,
   ) {
-    super(password, encryptedSeed, platformEncryptionTools, affinidiCommon, options, component)
+    super(password, encryptedSeed, platformCryptographyTools, options, component)
     this._userManagementService = createUserManagementService(options)
     this.cognitoUserTokens = cognitoUserTokens
   }
 
+  private static _isKeyParams(keyParamsOrOptions?: KeyParamsOrOptions): keyParamsOrOptions is KeyParams {
+    return !!keyParamsOrOptions && 'encryptedSeed' in keyParamsOrOptions
+  }
+
   private static _shouldCallAfterConfirmSignUp(keyParamsOrOptions?: KeyParamsOrOptions) {
-    return !keyParamsOrOptions
+    return !NetworkMemberWithCognito._isKeyParams(keyParamsOrOptions)
   }
 
   private static async _createKeyParams(
     options: ParsedOptions,
-    platformEncryptionTools: IPlatformEncryptionTools,
+    platformCryptographyTools: IPlatformCryptographyTools,
     shortPassword: string,
     keyParamsOrOptions?: KeyParamsOrOptions,
   ) {
-    if (!keyParamsOrOptions) {
-      return await NetworkMemberWithCognito._createSignUpKeys(shortPassword, options)
-    }
-
-    if ('encryptedSeed' in keyParamsOrOptions) {
+    if (NetworkMemberWithCognito._isKeyParams(keyParamsOrOptions)) {
       return keyParamsOrOptions
     }
 
-    return await NetworkMemberWithCognito._createKeyParamsForOptions(
-      platformEncryptionTools,
+    return await NetworkMemberWithCognito._createSignUpKeys(
+      options,
+      platformCryptographyTools,
       shortPassword,
       keyParamsOrOptions,
     )
   }
 
-  /**
-   * @deprecated Temporary implementation; refactor by 6.0 release (FTL-1707)
-   */
   private static _createKeyParamsOrOptionsValidator(keyParamsOrOptions?: KeyParamsOrOptions) {
-    if (!keyParamsOrOptions || 'encryptedSeed' in keyParamsOrOptions) {
+    if (!keyParamsOrOptions || NetworkMemberWithCognito._isKeyParams(keyParamsOrOptions)) {
```

Comment:
I think in static classes with static methods you can use `this`

Reply:
`this` meaning is subtly different, and TS will remind you of this as soon as you put any constraints on `this`.

We already have a bunch of these static methods in this class, they have to call other private static methods using class name instead of this (otherwise TS will not build the code).

In this specific case, considering that caller is private as well, and that it is never passed anywhere but only called directly, we could replace explicit class name with `this`, but that's probably not worth it, with all these qualifiers and with other methods using explicit class name anyway.